mousestyles/ultradian/__init__.py

```diff
@@ -0,0 +1,456 @@
+def strain_seasonal(strain, mouse, feature, bin_width, period_length):
+    """
+    Use seansonal decomposition model on the time series
+    of specified strain, mouse, feature and bin_width.
+    return the seasonal term and the plot of seasonal term
+    by mouse of a set of mouses in a strain
+
+    Parameters
+    ----------
+    strain: int
+        nonnegative integer indicating the strain number
+    mouse: list, set or tuple
+        nonnegative integer indicating the mouse number
+    feature: {"AS", "F", "M_AS", "M_IS", "W", "Distance"}
+        "AS": Active state probalibity
+        "F": Food consumed (g)
+        "M_AS": Movement outside homebase
+        "M_IS": Movement inside homebase
+        "W": Water consumed (g)
+        "Distance": Distance traveled
+    bin_width: int
+        number of minutes, the time interval for data aggregation
+    period_length: int
+        number of hours, usually the significant period
+        length indicated by Lomb-scargle model
+
+    Returns
+    -------
+    seasonal_all: numpy array containing the seasonal term for every
+        mouse indicated by the input parameter
+    seasonal_plot: plot of seasonal term by mouse
+
+    Examples
+    --------
+    >>> res = strain_seasonal(strain=0, mouse={0, 1, 2, 3}, feature="W",
+                              bin_width=30, period_length = 24)
+    """
+
+    if (not isinstance(strain, int)) or (strain < 0):
+        raise ValueError(
+            'Strain must be a non-negative integer')
+    if (not isinstance(mouse, int)) or (mouse < 0):
+        raise ValueError(
+            'Mouse value must be a non-negative integer')
+    if feature not in INTERVAL_FEATURES:
+        raise ValueError(
+            'Input value must in {"AS", "F", "M_AS", "M_IS", "W"}')
+    if (not isinstance(bin_width, int)) or bin_width < 0 or bin_width > 1440:
+        raise ValueError(
+            'Bin width (minutes) must be a non-negative integer below 1440')
+    if (not isinstance(period_length, int)) or period_length < 0:
+        raise ValueError(
+            'Peoriod length must be a non-negative integer')
+    # seasonal decomposition
+    seasonal_all = np.array([])
+    freq = int(period_length * 60 / bin_width)
+    for m in mouse:
+        res = seasonal_decomposition(
+            strain, m, feature, bin_width, period_length)
+        seasonal_all = np.append(seasonal_all, res.seasonal[0:freq])
+    seasonal_all = seasonal_all.reshape([len(mouse), -1])
+    # seasonal plot
+    seasonal_plot = plt.figure()
+    ax = seasonal_plot.add_subplot(1, 1, 1)
+    time = np.arange(0, period_length, bin_width / 60)
+    for i in np.arange(len(mouse)):
+        ax.plot(time, seasonal_all[i, :])
+    ax.legend(['mouse' + str(i) for i in np.arange(len(mouse))],
+              loc='upper right', prop={'size': 8})
+    plt.xlabel('Time')
+    plt.xlabel('Time')
+    plt.ylabel('Seasonal Term')
+    return seasonal_all, seasonal_plot
+
+
+def mix_strain(data, feature):
```

Comment:
I add "distance" to this raise ValueError but didn't work. Maybe I need to change INTERVAL_FEATURES at the very beginning of the file? But that will cause other functions' (like aggregate_interval) raising errors have problems. Maybe we should have an ALL_FEATURES set at the very beginning of the file?